gdal/ogr/ogrsf_frmts/lvbag/ogrlvbaglayer.cpp

```diff
@@ -316,6 +324,53 @@ void OGRLVBAGLayer::DataHandlerCbk( const char *data, int nLen )
         osElementString.append(data, nLen);
 }
 
+/************************************************************************/
+/*                              TouchLayer()                            */
+/************************************************************************/
+
+bool OGRLVBAGLayer::TouchLayer()
+{
+    // Only when the pool is used we register this layer as MRU.
+    if( poPool->GetSize() > 0 )
+        poPool->SetLastUsedLayer(this);
+
+    switch( eFileDescriptorsState )
+    {
+        case FD_OPENED:
+            return true;
+        case FD_CANNOT_REOPEN:
+            return false;
+        case FD_CLOSED:
+            break;
+    }
+
+    fp = VSIFOpenExL(osFilename, "rb", true);
+    if( !fp )
+    {
+        CPLError(CE_Warning, CPLE_AppDefined,
+            "Opening LV BAG extract failed : %s", osFilename.c_str());
+        eFileDescriptorsState = FD_CANNOT_REOPEN;
+        return false;
+    }
+    
+    eFileDescriptorsState = FD_OPENED; // TODO: Why?
```

Comment:
to answer the "TODO: Why" ? This prevents re-opening of the file descriptor the next time TouchLayer() is called.  We could just do that by testing fp state. But here the tri-state with FD_CANNOT_REOPEN prevents from trying to re-open when it already failed

Reply:
Was planning on testing fp is null, good point.